tests/test_pods.py

```diff
@@ -0,0 +1,168 @@
+    assert _mock_api_core_client.patch_namespaced_pod.call_args[1]["a"] == "test"
+
+
+async def test_read_namespaced_pod(kubernetes_credentials, _mock_api_core_client):
+    await read_namespaced_pod.fn(
+        pod_name="test_pod",
+        namespace="ns",
+        a="test",
+        kubernetes_credentials=kubernetes_credentials,
+    )
+    assert _mock_api_core_client.read_namespaced_pod.call_args[1]["name"] == "test_pod"
+    assert _mock_api_core_client.read_namespaced_pod.call_args[1]["namespace"] == "ns"
+    assert _mock_api_core_client.read_namespaced_pod.call_args[1]["a"] == "test"
+
+
+async def test_read_namespaced_pod_logs(kubernetes_credentials, _mock_api_core_client):
+    await read_namespaced_pod_logs.fn(
+        pod_name="test_pod",
+        container="test_container",
+        namespace="ns",
+        a="test",
+        kubernetes_credentials=kubernetes_credentials,
+    )
+    assert (
+        _mock_api_core_client.read_namespaced_pod_log.call_args[1]["name"] == "test_pod"
+    )
+    assert (
+        _mock_api_core_client.read_namespaced_pod_log.call_args[1]["namespace"] == "ns"
+    )
+    assert (
+        _mock_api_core_client.read_namespaced_pod_log.call_args[1]["container"]
+        == "test_container"
+    )
+    assert _mock_api_core_client.read_namespaced_pod_log.call_args[1]["a"] == "test"
+
+
+async def test_replace_namespaced_pod(kubernetes_credentials, _mock_api_core_client):
+    await replace_namespaced_pod.fn(
+        pod_name="test_pod",
+        body=V1Pod(**{"metadata": {"name": "test-pod"}}),
+        namespace="ns",
+        a="test",
+        kubernetes_credentials=kubernetes_credentials,
+    )
+    assert (
+        _mock_api_core_client.replace_namespaced_pod.call_args[1]["name"] == "test_pod"
+    )
+    assert (
+        _mock_api_core_client.replace_namespaced_pod.call_args[1]["namespace"] == "ns"
+    )
+    assert _mock_api_core_client.replace_namespaced_pod.call_args[1][
+        "body"
+    ].metadata == {"name": "test-pod"}
+    assert _mock_api_core_client.replace_namespaced_pod.call_args[1]["a"] == "test"
+
+
+@pytest.mark.parametrize(
+    "task_accepting_pod",
+    [create_namespaced_pod, patch_namespaced_pod, replace_namespaced_pod],
+)
+async def test_bad_v1_pod_kwargs(kubernetes_credentials, task_accepting_pod):
+    with pytest.raises(TypeError):
+        await task_accepting_pod.fn(
+            body=V1Pod(**{"random_not_real": "shabba-ranks"}),
+            kubernetes_credentials=kubernetes_credentials,
+        )
+
+
+async def test_read_pod_logs_custom_print_func_timeout(
+    kubernetes_credentials, mock_stream_timeout
+):
+    with pytest.raises(ApiException):
+        await read_namespaced_pod_logs.fn(
+            kubernetes_credentials=kubernetes_credentials,
+            pod_name="test_pod",
+            container="test_container",
+            namespace="ns",
+            print_func=print,
+        )
```

Comment:
Can you add a test where `print_func` is passed to `read_namespaced_pod_logs`, it prints logs, and doesn't time out?